autoPyTorch/data/utils.py

```diff
@@ -511,20 +524,19 @@ def reduce_dataset_size_if_too_large(
     """
 
     for method in methods:
+        if megabytes(X) > memory_allocation:
```

Comment:
Thanks for addressing my comment, but we should break the loop when we get `megabytes(X) <= memory_allocation`.
By doing it (see [here](https://github.com/automl/Auto-PyTorch/pull/398#discussion_r820704868)), we can reduce the nest.

Reply:
yeah I meant to do that. I'll fix this